HaloDAO/halo-rewards
# Feature/hdf 60 create update halo halo price

## Description of Changes
- created `estimatedHaloHaloApy`, `updateHaloHaloPrice()` functions in the Halohalo contract
- created the `latestHaloHaloPrice` public variable to access the current halohalo price and the last time it was updated based on the `HaloHaloPrice` struct
- added typescript support
- modified the config file to support ts
- modified the test to support ts
- separated tests for both contracts and renamed them
- added the hardhat gas reporter to check gas estimations
- added prettier (the one we use for all)


[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-60)
[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-70)

## How To Test
- reinstall packages (yarn or npm install)
- yarn run test / npm run test
- you should see all the tests passing and the gas estimation table after the tests are finished

### Developer Checklist:


### Reviewers Checklist:
